app/services/CommonTaskListService.scala

```diff
@@ -16,70 +16,79 @@
+                                      service: EmploymentOrchestrationService,
+                                      repository: JourneyAnswersRepository) extends Logging {
 
-  def get(taxYear: Int, nino: String, mtditid: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[TaskListSection] = {
+  private def getEmploymentTask(employments: AllEmploymentData,
+                                taxYear: Int,
+                                journeyAnswersOpt: Option[JourneyAnswers]): Option[Seq[TaskListSectionItem]] = {
 
-    service.getAllEmploymentData(nino, taxYear, mtditid).map {
-      case Left(_) => None
-      case Right(employments) => getTask(employments, taxYear)
-    }.map { optionalTask =>
-      TaskListSection(SectionTitle.EmploymentTitle, optionalTask)
-    }
-  }
-
-  private def getTask(employments: AllEmploymentData, taxYear: Int): Option[Seq[TaskListSectionItem]] = {
-
-    val employmentUrl: String = s"${appConfig.employmentFEBaseUrl}/update-and-submit-income-tax-return/employment-income/$taxYear/employment-summary"
+    val baseUrl = s"${appConfig.employmentFEBaseUrl}/update-and-submit-income-tax-return"
+    val employmentUrl: String = s"$baseUrl/employment-income/$taxYear/employment-summary"
 
     def employmentTask(taskStatus: TaskStatus): Option[Seq[TaskListSectionItem]] =
       Some(Seq(TaskListSectionItem(TaskTitle.PayeEmployment, taskStatus, Some(employmentUrl))))
 
-    val hmrcSubmittedOn: Seq[Long] = {
-        employments.hmrcEmploymentData.map { employment =>
-          val submittedOn: String = employment.hmrcEmploymentFinancialData.flatMap(_.employmentData.map(_.submittedOn))
-            .getOrElse(Instant.MIN.toString)
-
-          getSubmittedOnEpoch(submittedOn)
-        }
-      }
-
-    val customerSubmittedOn: Seq[Long] =
-      employments.hmrcEmploymentData.map { employment =>
-        val submittedOn: String  = employment.customerEmploymentFinancialData.flatMap(_.employmentData.map(_.submittedOn))
-          .getOrElse(Instant.MIN.toString)
+    def toSubmittedOn(financialDataOpt: Option[EmploymentFinancialData]): Instant =
+      financialDataOpt.flatMap(
+        _.employmentData.map(
+          data => Instant.parse(data.submittedOn)
+        )
+      ).getOrElse(Instant.MIN)
 
-        getSubmittedOnEpoch(submittedOn)
-      }
+    def isHmrcLatest = employments.hmrcEmploymentData.exists(employment => {
+      val hmrcHeldSubmittedOn = toSubmittedOn(employment.hmrcEmploymentFinancialData)
+      val customerHeldSubmittedOn = toSubmittedOn(employment.customerEmploymentFinancialData)
+      hmrcHeldSubmittedOn.isAfter(customerHeldSubmittedOn)
+    })
 
     val hasCustomerData: Boolean = employments.customerEmploymentData.exists(_.employmentData.nonEmpty)
+    val hasHmrcData: Boolean = employments.hmrcEmploymentData.exists(_.hmrcEmploymentFinancialData.nonEmpty)
+
+    (hasHmrcData, hasCustomerData, journeyAnswersOpt) match {
```

Comment:
Should this logic be as below, because if customer data is not present and HMRC is present then we do not need to compare, HMRC data should be treated latest. Lets discuss this on call. Also not sure about returning None at the end `    (hasHmrcData, hasCustomerData, journeyAnswersOpt) match {
      case (true, true, _) if isHmrcLatest => employmentTask(CheckNow)
      case (true, _, _) => employmentTask(CheckNow)
      case (_, true, _) => employmentTask(if(appConfig.sectionCompletedQuestionEnabled) InProgress else Completed)
      case (_, _, Some(JourneyAnswers(_, _, _, data, _))) =>
        val status = data.value("status").validate[TaskStatus].asOpt.getOrElse {
          logger.info("[CommonTaskListService][getStatus] status stored in an invalid format, setting as 'Not yet started'.")
          NotStarted
        }

        employmentTask(status)
      case (_, _, _) => None
    }`